docker/pypi/wmagent/init.sh

```diff
@@ -329,12 +329,27 @@ EOF
     echo "-----------------------------------------------------------------------"
 }
 
-check_docker_init() {
-    # A function to check all previously populated */.dockerInit files
-    # from all previous steps and compare them with the /data/.dockerBuildId
+check_wmagnet_init() {
+    # A function to check all previously populated */.init<step> files
+    # from all previous steps and compare them with the /data/.wmaBuildId
     # if all do not match we cannot continue - we consider configuration/version
     # mismatch between the host and the container
 
+    # NOTE: On every step we need to check the .init<step> file content. We always compare
+    #       the current $WMA_BUILD_ID with the one previously initialized at the host.
+    #       The WMA_BUILD_ID is generated during the execution of `install.sh`
+    #       at build time (for docker images) or deploy time for virtual env
+    #       There are three levels of comparison we can make:
+    #       * if we want to trigger re-initialization on any WMAgent image rebuild,
+    #         then the $WMA_BUILD_ID should contain a sha256sum of a random variable
+    #       * if we want to trigger re-initialization only on new WMAgent tag builds,
+    #         then the $WMA_BUILD_ID should contain a sha256sum of whole $WMA_TAG
+    #       * if we want to trigger re-initialization only on version change (not on patches),
```

Comment:
Then I would suggest making the bullet points above as an ordered list, just so whoever is working with this code can understand that there are multiple ways that this could have been implemented. Nonetheless, I still think the most important is to say what IS the expected functionality and behavior.

Reply:
ok